cap/modules/services/views/adl.py

```diff
@@ -0,0 +1,313 @@
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# CERN Analysis Preservation Framework is distributed in the hope that it will
+# be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with CERN Analysis Preservation Framework; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""CAP ADL ingestion service views."""
+
+from typing import List, Optional
+from dataclasses import dataclass, field
+import re
+from flask import jsonify, abort, current_app
+
+from invenio_files_rest.models import FileInstance, ObjectVersion
+from invenio_pidstore.resolver import Resolver
+from invenio_pidstore.errors import PIDDoesNotExistError
+
+from . import blueprint
+from cap.modules.access.utils import login_required
+from cap.modules.deposit.api import CAPDeposit
+
+
+TABLE_HEADER = '# val    err-     err+     xmin     xmax'
+
+
+@dataclass(frozen=True)
+class Line:
+    text: str # noqa
+
+
+@dataclass
+class Block:
+    block_type: Optional[str] = None # noqa
+    lines: List[Line] = field(default_factory=list) # noqa
+
+
+@dataclass
+class Keyword:
+    block_type: Optional[str] = None # noqa
+    line: Line = None # noqa
+
+
+def _collect_lines(file_handler):
+    """Return a line generator for the block/keyword."""
+    _COMMENT = '#'
+    for line in file_handler:
+        line = line.decode('utf-8')
+        text = line.strip()
+        if not text:
+            continue
+        elif text.startswith(TABLE_HEADER):
+            yield Line(text)
+        elif not text.startswith(_COMMENT):
+            # Skip comment lines
+            yield Line(text)
+
+
+def _collect_blocks(lines):
+    """Return a block/keyword generator."""
+    _block_object = Block()
+    for line in lines:
+        _block_type = line.text.split()[0]
+        if _regex_block_match(_block_type):
+            if _block_object.lines:
+                yield _block_object
+            _block_object = Block(block_type=_block_type)
+        if _regex_keyword_match(_block_type, define=True):
+            if _block_object.lines:
+                yield _block_object
+            _keyword_object = Keyword(block_type=_block_type, line=line)
+            yield _keyword_object
+        if _block_object:
+            _block_object.lines.append(line)
+    if _block_object and _block_object.lines:
+        yield _block_object
+
+
+def _regex_block_match(_block_type):
+    _block_type_regexp = {
+        'object': re.compile(r'object'),
```

Comment:
For the use case, since we skip keywords that are not mentioned in the ADL twiki. I can't find any reference to this type of block, shouldn't we skip this as well?

Reply:
How do you handle it now? what is the output for this example part ?

Reply:
This is the output for these objects:

```
{
        {
            "object": "leptons : Union( electrons , muons)"
        },
        {
            "object": "vetoleptons : Union( vetoelectrons, vetomuons)"
        },
}
```

Reply:
Ok lets fix it..this is not correct. Either throw an exception or just handle `Union()`cases together with other regex